app/components/ui/select.tsx

```diff
@@ -0,0 +1,137 @@
+import React, {FC, PropsWithChildren, useState} from 'react'
+import {Box, Button} from '@effortless-ui'
+import {IconCheck} from '@tabler/icons-react'
+
+import {ExpandableArrow} from '@/app/components/ui'
+import {useOutsideElementClickHandler} from '@/app/hooks'
+import {theme} from '@/app/styles'
+
+export interface ISelectOption {
+  label: string
+  value: string
+}
+
+export interface ISelectProps {
+  onChange?(
+    value: string[],
+    selected: {
+      selectedLabel: string
+      selectedValue: string
+    },
+  ): void
+  options: ISelectOption[]
+  title?: string
+  value?: string[]
+}
+
+export const Select: FC<PropsWithChildren<ISelectProps>> = ({children, onChange, options, title, value}) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [isMouseOver, setIsMouseOver] = useState<boolean>(false)
+  const ref = useOutsideElementClickHandler(() => setIsOpen(false))
+
+  return (
+    <Box cs={{label: 'Select', position: 'relative'}} ref={ref}>
+      <Button
+        onClick={() => setIsOpen(!isOpen)}
+        onMouseEnter={() => setIsMouseOver(true)}
+        onMouseLeave={() => setIsMouseOver(false)}
+        cs={{
+          display: 'inline-flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+          columnGap: theme.spacing.xs,
+          px: theme.spacing.s,
+          py: theme.spacing.xs,
+          color: theme.color.text,
+          border: '1px solid',
+          borderColor: theme.color.text,
+          borderRadius: theme.radii.s,
+          backgroundColor: isOpen ? theme.color.white : 'transparent',
+          cursor: 'pointer',
+          transition: 'color 200ms, background-color 200ms, border-color 200ms',
+          '&:hover': {
+            color: theme.color.primary,
+            borderColor: theme.color.primary,
+            backgroundColor: theme.color.white,
+          },
+        }}
+      >
+        {title ?? children}
+        <ExpandableArrow direction={isOpen ? 'up' : 'down'} color={isMouseOver ? theme.color.primary : theme.color.text} size={22} />
+      </Button>
+      <Box
+        tag="ul"
+        composition={['semanticList']}
+        cs={{
+          position: 'absolute',
+          top: '100%',
+          left: 0,
+          mt: theme.spacing.xxs,
+          display: 'flex',
+          rowGap: theme.spacing.s,
+          flexDirection: 'column',
+          p: theme.spacing.s,
+          border: '1px solid',
+          borderColor: theme.color.text,
+          borderRadius: theme.radii.s,
+          backgroundColor: theme.color.white,
+          opacity: isOpen ? '1' : '0',
+          pointerEvents: isOpen ? 'auto' : 'none',
+          transform: isOpen ? 'translateY(0)' : `translateY(-${theme.spacing.xxs}px)`,
+          zIndex: 3,
```

Comment:
nie chcemy w theme trzymać z-indexów? zamiast z palca ich podawać?

Reply:
Z doświadczenia nigdy nie widziałem korzyści z trzymania indeksów zewnętrznym pliku - nie poprawia to czytelności, a wprowadza dodatkową warstwę skomplikowania. Według mnie nie na takiej potrzeby.